pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -1880,77 +1924,77 @@ def test_dti_with_offset_series(self, tz_naive_fixture, names):
             res3 = dti - other
         tm.assert_series_equal(res3, expected_sub)
 
-    def test_dti_add_offset_tzaware(self, tz_aware_fixture, box_with_array):
-        # GH#21610, GH#22163 ensure DataFrame doesn't return object-dtype
-        timezone = tz_aware_fixture
-        if timezone == 'US/Pacific':
-            dates = date_range('2012-11-01', periods=3, tz=timezone)
-            offset = dates + pd.offsets.Hour(5)
-            assert dates[0] + pd.offsets.Hour(5) == offset[0]
-
-        dates = date_range('2010-11-01 00:00',
-                           periods=3, tz=timezone, freq='H')
-        expected = DatetimeIndex(['2010-11-01 05:00', '2010-11-01 06:00',
-                                  '2010-11-01 07:00'], freq='H', tz=timezone)
-
-        # FIXME: these raise ValueError with transpose=True
-        dates = tm.box_expected(dates, box_with_array, transpose=False)
-        expected = tm.box_expected(expected, box_with_array, transpose=False)
 
-        # TODO: parametrize over the scalar being added?  radd?  sub?
-        offset = dates + pd.offsets.Hour(5)
-        tm.assert_equal(offset, expected)
-        offset = dates + np.timedelta64(5, 'h')
-        tm.assert_equal(offset, expected)
-        offset = dates + timedelta(hours=5)
-        tm.assert_equal(offset, expected)
-
-
-@pytest.mark.parametrize('klass', [Series, DatetimeIndex])
-def test_dt64_with_offset_array(klass):
+def test_dt64_with_offset_array(box_with_array):
     # GH#10699
     # array of offsets
-    box = Series if klass is Series else pd.Index
-
-    s = klass([Timestamp('2000-1-1'), Timestamp('2000-2-1')])
+    s = DatetimeIndex([Timestamp('2000-1-1'), Timestamp('2000-2-1')])
+    s = tm.box_expected(s, box_with_array)
 
-    with tm.assert_produces_warning(PerformanceWarning,
+    warn = PerformanceWarning if box_with_array is not pd.DataFrame else None
+    with tm.assert_produces_warning(warn,
                                     clear=[pd.core.arrays.datetimelike]):
-        result = s + box([pd.offsets.DateOffset(years=1),
+        other = pd.Index([pd.offsets.DateOffset(years=1),
                           pd.offsets.MonthEnd()])
-        exp = klass([Timestamp('2001-1-1'), Timestamp('2000-2-29')])
+        other = tm.box_expected(other, box_with_array)
```

Comment:
this file getting kind of hefty. ok with making a subdir if it helps, e.g.

pandas/tests/arithmetic/datetime64/test_offsets....etc.

Reply:
I'm hopeful we'll be able to de-duplicate a bunch of these (many moved from series/frame/index tests) and slim it down.  If not then subdirs may be the way to go.